test/qpy_compat/test_qpy.py

```diff
@@ -330,6 +333,10 @@ def generate_evolution_gate():
     from qiskit.circuit.library import PauliEvolutionGate
     from qiskit.synthesis import SuzukiTrotter
 
+    with warnings.catch_warnings():
+        warnings.simplefilter("ignore", DeprecationWarning)
+        from qiskit.opflow import I, Z
+
     synthesis = SuzukiTrotter()
     evo = PauliEvolutionGate([(Z ^ I) + (I ^ Z)] * 5, time=2.0, synthesis=synthesis)
```

Comment:
Here, we are again not testing anything QPY-related using opflow, the `PauliEvolutionGate` tests already have both cases, so maybe we could also just get rid of the dependency. In that case, the alternative would be:

```suggestion
    synthesis = SuzukiTrotter()
    op = SparsePauliOp.from_list([('ZI', 1), ('IZ', 1)])
    evo = PauliEvolutionGate([op] * 5, time=2.0, synthesis=synthesis)
```